server/src/db/system_tables.rs

```diff
@@ -142,21 +143,28 @@ fn from_partition_summaries(partitions: Vec<PartitionSummary>) -> Result<RecordB
                 partition_key.append_value(&partition.key)?;
                 table_name.append_value(&table.name)?;
                 column_name.append_value(&column.name)?;
-                count.append_value(column.count())?;
+                column_type.append_value(column.type_name())?;
+                if let Some(t) = &column.influxdb_type {
+                    influxdb_type.append_value(t.as_str())?;
+                } else {
+                    influxdb_type.append_null()?;
+                }
             }
         }
     }
 
     let partition_key = partition_key.finish();
     let table_name = table_name.finish();
     let column_name = column_name.finish();
-    let count = count.finish();
+    let column_type = column_type.finish();
+    let influxdb_type = influxdb_type.finish();
 
-    RecordBatch::try_from_iter(vec![
-        ("partition_key", Arc::new(partition_key) as ArrayRef),
-        ("table_name", Arc::new(table_name)),
-        ("column_name", Arc::new(column_name)),
-        ("count", Arc::new(count)),
+    RecordBatch::try_from_iter_with_nullable(vec![
+        ("partition_key", Arc::new(partition_key) as ArrayRef, false),
+        ("table_name", Arc::new(table_name), false),
+        ("column_name", Arc::new(column_name), false),
+        ("column_type", Arc::new(column_type), false),
+        ("influxdb_type", Arc::new(influxdb_type), false),
```

Comment:
Hm, above has:

```
} else {
   influxdb_type.append_null()?;
}
```

so if I'm understanding the arrow concepts correctly, shouldn't this line have `true` for the third element in the tuple to say that `influxdb_type` *can* be null?

```suggestion
        ("influxdb_type", Arc::new(influxdb_type), true),
```

Reply:
I think you are correct. Thank you for the spot. 👍